src/styles/mixins/_inputs.scss

```diff
@@ -0,0 +1,142 @@
+/* This file contains mixins with styles used in all types of inputs */
+
+/* Common elements */
+
+@mixin inputHelpText {
+  color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+  font-family: var(--kui-font-family-text, $kui-font-family-text);
+  font-size: var(--kui-font-size-20, $kui-font-size-20);
+  font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
+  line-height: var(--kui-line-height-20, $kui-line-height-20);
+}
+
+/* Text inputs */
+
+@mixin inputDefaults {
+  -webkit-appearance: none; // need this to allow box-shadow to apply on mobile
+  background-color: var(--kui-color-background, $kui-color-background);
+  border: 0;
+  border-radius: var(--kui-border-radius-30, $kui-border-radius-30);
+  box-shadow: var(--kui-shadow-border, $kui-shadow-border);
+  color: var(--kui-color-text, $kui-color-text);
+  font-family: var(--kui-font-family-text, $kui-font-family-text);
+  font-size: 16px; // needs to be at least 16px to prevent automatic zoom in on focus on mobile
+  font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
+  line-height: var(--kui-line-height-40, $kui-line-height-40);
+  outline: none;
+  padding: var(--kui-space-40, $kui-space-40) $kInputPaddingX;
+  transition: box-shadow $kongponentsTransitionDurTimingFunc;
+  width: 100%;
+
+  @media (min-width: $kui-breakpoint-mobile) {
+    font-size: var(--kui-font-size-30, $kui-font-size-30);
+  }
+
+  &::placeholder {
+    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+  }
+}
+
+@mixin inputHover {
+  box-shadow: var(--kui-shadow-border-primary-weak, $kui-shadow-border-primary-weak);
+}
+
+@mixin inputFocus {
+  box-shadow: var(--kui-shadow-border-primary, $kui-shadow-border-primary), var(--kui-shadow-focus, $kui-shadow-focus);
+}
+
+// shared styles between disabled and readonly states
+@mixin inputDisabledReadOnly {
+  background-color: var(--kui-color-background-disabled, $kui-color-background-disabled);
+  box-shadow: var(--kui-shadow-border-disabled, $kui-shadow-border-disabled);
+  color: var(--kui-color-text-disabled, $kui-color-text-disabled);
+}
+
+@mixin inputDisabled {
+  @include inputDisabledReadOnly;
+
+  cursor: not-allowed;
+}
+
+@mixin inputReadOnly {
+  @include inputDisabledReadOnly;
+}
+
+/* Radio and checkbox inputs */
```

Comment:
question/suggestion: since radios and checkboxes are a specific thing, should they maybe have their own mixins file, or are the rules somewhat "intertwined" and it makes sense to leave in here?

Reply:
I think they belong together intuitively. They aren't intertwined at the moment but soon we will add more styles for other types of inputs (select, multiselect) and it will make more sense to view this file as "all things related to input elements". Otherwise I struggle to think of a file name because `_radios-checkboxes.scss` sounds a bit weird?

Reply:
I definitely think the select/multislect type inputs should go into a separate file -- this file is already pretty large 😬

Reply:
Eh I guess. `_radios-checkboxes.scss` it is then.

Reply:
`_input-radio-checkbox.scss`? 😬

Reply:
Then you can also do `_input-select.scss` `_input-multiselect.scss`

Reply:
Done